modules/src/ics07_tendermint/client_state.rs

```diff
@@ -126,14 +135,16 @@ impl TryFrom<RawClientState> for ClientState {
                 raw.frozen_height
                     .ok_or_else(|| Kind::InvalidRawClientState.context("missing frozen height"))?,
             ),
+            allow_update_after_expiry: raw.allow_update_after_expiry,
+            allow_update_after_misbehaviour: raw.allow_update_after_misbehaviour,
         })
     }
 }
 
 impl From<ClientState> for RawClientState {
     fn from(value: ClientState) -> Self {
         RawClientState {
-            chain_id: value.chain_id,
+            chain_id: value.clone().chain_id,
```

Comment:
It's probably less expensive to clone `value.chain_id` rather than the whole `value` (if it works at all of course)

```suggestion
            chain_id: value.chain_id.clone(),
```